genesyscloud/outbound_ruleset/resource_genesyscloud_outbound_ruleset.go

```diff
@@ -131,3 +144,59 @@ func deleteOutboundRuleset(ctx context.Context, d *schema.ResourceData, meta int
 		return retry.RetryableError(fmt.Errorf("Outbound Ruleset %s still exists", d.Id()))
 	})
 }
+
+// filterOutboundRuleset filters rulesets by removing the ones that are referencing skills that no longer exist in GC
+func filterOutboundRulesets(ruleSets []platformclientv2.Ruleset, skillMap resourceExporter.ResourceIDMetaMap) ([]platformclientv2.Ruleset, diag.Diagnostics) {
+	var filteredRuleSets []platformclientv2.Ruleset
+	log.Printf("Filtering outbound rule sets")
+
+RuleSetLoop:
+	for _, ruleSet := range ruleSets {
+		for _, rule := range *ruleSet.Rules {
+			// look through rule actions to check if the referenced skills exist in our skill map or not
+			for _, action := range *rule.Actions {
+				if action.ActionTypeName != nil && strings.ToLower(*action.ActionTypeName) == "set_skills" && action.Properties != nil {
+					for id, value := range *action.Properties {
```

Comment:
This block is a good candidate to break out into a a set of smaller methods and then write unit tests.